### PR TITLE
MacOS support : hack around SIP to load libc

### PR DESCRIPTION
**Pull request by catenacyber**
see https://stackoverflow.com/questions/32905322/oserror-dlopenlibsystem-dylib-6-image-not-found

For #110

---
This change is [Reviewable](https://reviewable.io/reviews/trailofbits/manticore/862)

### CODE REVIEW

**Comment by mossberg** [comment]
hi @catenacyber can you provide me with a little additional context around the PR? what was the situation you ran into that motivated this change? i have very little experience trying to run manticore on os x :)

**Comment by catenacyber** [comment]
After my latest git pull, manticore did not start anymore and I had a python traceback.
Looking at it, it was pretty obvious that it failed trying to open libc.dylib
So I looked for it and found it in /usr/lib/
Maybe something smarter can be done to find the right path...

With this hack, I was able to run manticore again and to work on #861 

Manticore is not the only one with this problem, see for instance https://github.com/ffi/ffi/issues/461

File: manticore/utils/mappings.py [hunk 1]
```diff
@@ -10,7 +10,10 @@
 if osname == "darwin" or osname.startswith("linux"):
 
     if osname == "darwin":
-        libc = ctypes.cdll.LoadLibrary("libc.dylib")
+        try:
```
Comment (disconnect3d) [inline manticore/utils/mappings.py]: It looks like we can refactor above ifs into:
```
if osname == 'darwin':
    # ...
elif osname.startswith('linux'):
    # ...
```

**Comment by yan** [comment]
@catenacyber , sorry for the late review; can you update your branch to current `master`? That file has significantly changed, this PR would likely need to be reworked or might not be necessary anymore.

**Comment by catenacyber** [comment]
It looks like it is working now without my patch, so we should just close this pull request...

**Pull request closed without merging**